include/ex_actor/internal/actor_ref_deserialization.h

```diff
@@ -0,0 +1,131 @@
+        return std::make_pair(field, ix);
+      }
+      ++ix;
+    }
+    return std::optional<std::pair<capnp::StructSchema::Field, size_t>>();
+  }
+};
+
+template <class T, class... Ps>
+auto read(const InputVarType& _obj, const ex_actor::internal::ActorRefDeserializationInfo& info) {
+  const ActorRefReader r {info};
+  return rfl::parsing::Parser<ActorRefReader, Writer, T, Processors<SnakeCaseToCamelCase, Ps...>>::read(r, _obj);
+}
+
+template <class T, class... Ps>
+Result<internal::wrap_in_rfl_array_t<T>> read(const concepts::ByteLike auto* bytes, size_t size,
+                                              const Schema<T>& schema,
+                                              const ex_actor::internal::ActorRefDeserializationInfo& info) {
+  const auto array_ptr = kj::ArrayPtr<const kj::byte>(internal::ptr_cast<const kj::byte*>(bytes), size);
+  auto input_stream = kj::ArrayInputStream(array_ptr);
+  auto message_reader = capnp::PackedMessageReader(input_stream);
+  const auto root_name = get_root_name<std::remove_cv_t<T>, Ps...>();
+  const auto root_schema = schema.value().getNested(root_name.c_str());
+  const auto input_var = InputVarType {message_reader.getRoot<capnp::DynamicStruct>(root_schema.asStruct())};
+  return read<T, Ps...>(input_var, info);
+}
+}  // namespace capnproto
+
+namespace parsing {
+template <class ProcessorsType, class U>
+struct Parser<capnproto::ActorRefReader, capnproto::Writer, ex_actor::internal::ActorRef<U>, ProcessorsType> {
+  using Reader = capnproto::ActorRefReader;
+  using InputVarType = typename Reader::InputVarType;
+
+  static Result<ex_actor::internal::ActorRef<U>> read(const Reader& reader, const InputVarType& var) noexcept {
+    using Type = typename Reflector<ex_actor::internal::ActorRef<U>>::ReflType;
+    auto actor_ref =
+        Parser<capnproto::Reader, capnproto::Writer, ex_actor::internal::ActorRef<U>, ProcessorsType>::read(reader, var)
+            .value();
+    const auto& info = reader.info;
+    actor_ref.SetLocalRuntimeInfo(info.this_node_id, info.actor_look_up_fn(actor_ref.GetActorId()),
+                                  info.message_broker);
+    return actor_ref;
+  }
+
+  template <class Parent>
+  static void write(const capnproto::Writer& w, const ex_actor::internal::ActorRef<U>& ref, const Parent& parent) {
+    Parser<capnproto::Reader, capnproto::Writer, ex_actor::internal::ActorRef<U>, ProcessorsType>::write(w, ref,
+                                                                                                         parent);
+  }
+};
+
+}  // namespace parsing
+
+}  // namespace rfl
+
+namespace ex_actor::internal::serde {
+template <class T>
+T Deserialize(const uint8_t* data, size_t size, const ActorRefDeserializationInfo& info) {
+  return rfl::capnproto::read<T>(data, size, GetCachedSchema<T>(), info).value();
+}
+
+template <auto kFn>
+auto DeserializeFnArgs(const uint8_t* data, size_t size, const ActorRefDeserializationInfo& info) {
```

Comment:
Can we keep DeserializeFnArgs in serialization.h? Because this seems a common function.